metrics_sdk/lib/opentelemetry/sdk/metrics/export.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module Export
+        ExportError = Class.new(OpenTelemetry::Error)
+
+        # The operation finished successfully.
+        SUCCESS = 0
+
+        # The operation finished with an error.
+        FAILURE = 1
+
+        # The operation timed out.
+        TIMEOUT = 2
```

Comment:
Is this part of the spec? For tracing, it's only used as an extra result code for `force_flush` and `shutdown`. The spec is handwavy about whether it should be a code (it could equally be an exception).

Reply:
I got similar hand wavy interpretations, but maybe I'm being too liberal with that and should not codify the timeout result.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#collect

>Collect does not have any required parameters, however, [OpenTelemetry SDK](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#sdk) authors MAY choose to add parameters (e.g. callback, filter, timeout). [OpenTelemetry SDK](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#sdk) authors MAY choose the return value type, or do not return anything.

Reply:
Re-reading:

> Collect SHOULD provide a way to let the caller know whether it succeeded, failed or timed out

So I think this is fine.